### PR TITLE
Add cropping large words around tag

### PR DESCRIPTION
**Pull request by yarynka97**
---
### Linked JIRA Tickets

<sub>&nbsp;</sub> <a href="https://gemshelf.atlassian.net/browse/GEM-6379">GEM-6379&nbsp;Search Hits/Gem Page: Collapsed search hit trimmed completely if there is long word in hit</a>

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -69,6 +69,13 @@ function getTextBetweenHighlightBoundaries({text, OPEN_TAG, CLOSE_TAG}) {
 
 function trimTextUntilSize(text, maxLengthAround, omission) {
   const wordsInput = text.split(' ').reverse();
+  const firstWord = [...wordsInput[0]].reverse().join('');
```
Comment (KnupMan) [inline index.js]: Why spread array into array? It will be the same

Comment (yarynka97) [reply index.js]: I'm spreading first item of array, which is string.

File: index.js [hunk 2]
```diff
@@ -110,3 +125,10 @@ function getWordsUntilLength(words, maxLength) {
 
   return wordsOutput;
 }
+
+function croppWordToGivenSize(word, outputLength) {
+  return word
+    .split('')
+    .slice(0, outputLength)
```
Comment (vladgolubev) [inline index.js]: By the way, `.slice()` works on strings as well, so no need to `split` & `join`

image

File: index.js [hunk 3]
```diff
@@ -110,3 +125,10 @@ function getWordsUntilLength(words, maxLength) {
 
   return wordsOutput;
 }
+
+function croppWordToGivenSize(word, outputLength) {
```
Comment (vladgolubev) [inline index.js]: typo `cropp` -> `crop`

File: index.js [hunk 4]
```diff
@@ -90,7 +94,7 @@ function trimTextUntilSizeFromEnd(text, maxLengthAround, omission) {
   const firstWord = wordsInput[0];
 
   if (firstWord.length > maxLengthAround) {
-    const cropedWord = cropWordToGivenSize(firstWord, maxLengthAround);
+    const cropedWord = [...firstWord].slice(0, maxLengthAround).join('');
```
Comment (KnupMan) [inline index.js]: So let's you just `slice` on word as a string, don't need to spread it to array

**Pull request merged**